tests/components/CopyPaste/CopyPaste.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { IQueryResult, $$, l } from 'coveo-search-ui';
+import { createSandbox, SinonSandbox, SinonStub } from 'sinon';
+import { Mock, Fake } from 'coveo-search-ui-tests';
+import { CopyPaste } from '../../../src/components/CopyPaste/CopyPaste';
+
+describe('ResultAction', () => {
+    let sandbox: SinonSandbox;
+
+    let componentSetup: Mock.IBasicComponentSetup<CopyPaste>;
+    let result: IQueryResult;
+    let element: HTMLElement;
+    let testComponent: CopyPaste;
+    let testOptions: Mock.AdvancedComponentSetupOptions;
+    let writeTextStub: SinonStub<[string], Promise<void>>;
+
+    beforeAll(() => {
+        sandbox = createSandbox();
+    });
+
+    beforeEach(async () => {
+        result = Fake.createFakeResult();
+        element = document.createElement('div');
+        document.body.append(element);
+        testOptions = new Mock.AdvancedComponentSetupOptions(element);
+
+        result.title = 'some title';
+        result.clickUri = 'some uri';
+
+        componentSetup = Mock.advancedResultComponentSetup(CopyPaste, result, testOptions);
+        testComponent = componentSetup.cmp;
+        writeTextStub = sandbox.stub(navigator.clipboard, 'writeText').resolves();
+    });
+
+    afterEach(() => {
+        sandbox.reset();
+        sandbox.restore();
+        $$(document.body)
+            .children()
+            .forEach(el => el.remove());
+    });
+
+    describe('on click', () => {
+        it('should paste the result in the clipboard', async () => {
+            testComponent.element.click();
+            expect(writeTextStub.called).toBeTrue();
+        });
+    });
+
+    describe('on mouseleave', () => {
+        it('should change the tooltip text for "Copy"', () => {
+            testComponent.tooltip.innerText = 'patate';
```

Comment:
I think you souldn't expose `.tooltip`. I guess you should validate this behavior using a functional test instead. If you prefer to use a unit test, maybe we could expose an event instead.

Reply:
You can't use Functionnal tests because of some issues in cypress regarding clipboard and hovering